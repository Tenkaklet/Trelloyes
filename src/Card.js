import React , { Component } from 'react';



class Card extends Component {

    removeCard = (e) => {
        this.props.onDelete(this.props.cardIndex);
        
    }




    render() {

        return (
            <div className="Card">
                <h3>{this.props.title}</h3>
                <p>{this.props.content}</p>
                <button onClick={() => this.removeCard(this.props.cardIndex)}>Delete</button>
            </div>
        )
    }
}

export default Card;