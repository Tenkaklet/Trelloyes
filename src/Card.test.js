import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';

describe('The Cards Component', () => {
    it('Renders without Crashing',  () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card title="title test" content="content" />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
    it('Renders the UI as expected', () => {
        const tree = renderer.create(<Card title="title test" content="content" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});