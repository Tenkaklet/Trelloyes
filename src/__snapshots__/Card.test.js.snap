// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Cards Component Renders the UI as expected 1`] = `
<div
  className="Card"
>
  <h3>
    title test
  </h3>
  <p>
    content
  </p>
</div>
`;
